spring-cloud-gateway-core/src/test/java/org/springframework/cloud/gateway/filter/GatewayMetricFilterTests.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2013-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.springframework.cloud.gateway.filter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.gateway.config.GatewayProperties;
+import org.springframework.cloud.gateway.test.BaseWebClientTests;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.reactive.function.client.ClientResponse;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@DirtiesContext
+public class GatewayMetricFilterTests extends BaseWebClientTests {
+
+	private static final String REQUEST_METRICS_NAME = "gateway.requests";
+	
+	@Autowired
+	private GatewayProperties properties;
+
+	@Autowired
+	private MeterRegistry meterRegistry;
+
+	@Test
+	public void gatewayRequestsMeterFilterHasTags() throws InterruptedException {
+		assertThat(this.properties.getDefaultFilters()).isNotEmpty();
+
+		ClientResponse clientResponse = webClient.get().uri("/headers").exchange().block();
+		assertEquals(clientResponse.statusCode(), HttpStatus.OK);
+		Thread.sleep(2000); // simulate scrape interval
```

Comment:
Unnecessary. The count will be nonzero as soon as the request is executed. `SimpleMeterRegistry`'s default mode is cumulative.

Reply:
If I don't put some sleep in then sometimes then sometimes the metric is not yet available in the registry.

Reply:
@ryanjbaxter Is there a better test facility for this?